src/main/java/com/google/api/generator/engine/ast/GeneralForStatement.java

```diff
@@ -43,60 +35,85 @@ public void accept(AstNodeVisitor visitor) {
   }
 
-  public static Builder builder() {
+  private static Builder builder() {
     return new AutoValue_GeneralForStatement.Builder().setBody(Collections.emptyList());
   }
 
   @AutoValue.Builder
-  public abstract static class Builder {
-    public abstract Builder setInitializationExpr(Expr initializationExpr);
-
-    public abstract Builder setBody(List<Statement> body);
-
-    // Private.
-    abstract Builder setLocalVariableExpr(VariableExpr variableExpr);
-
-    abstract Builder setMaxSizeExpr(Expr maxSizeExpr);
-
-    abstract VariableExpr localVariableExpr();
+  abstract static class Builder {
+    // Private setter.
+    abstract Builder setInitializationExpr(AssignmentExpr initializationExpr);
+    // Private setter.
+    abstract Builder setTerminationExpr(Expr terminationExpr);
+    // Private setter.
+    abstract Builder setUpdateExpr(Expr incrementExpr);
+    // Private setter.
+    abstract Builder setBody(List<Statement> body);
 
     abstract GeneralForStatement autoBuild();
 
     // Type-checking will be done in the sub-expressions.
-    public GeneralForStatement build() {
-      VariableExpr varExpr = localVariableExpr();
-      Preconditions.checkState(
-          varExpr.scope().equals(ScopeNode.LOCAL),
-          String.format(
-              "Variable %s in a general for-loop cannot have a non-local scope",
-              varExpr.variable().identifier().name()));
-      Preconditions.checkState(
-          !varExpr.isStatic() && !varExpr.isFinal(),
-          String.format(
-              "Variable %s in a general for-loop cannot be static or final",
-              varExpr.variable().identifier().name()));
-      setInitializationExpr(
-          AssignmentExpr.builder()
-              .setVariableExpr(varExpr.toBuilder().setIsDecl(true).build())
-              .setValueExpr(
-                  ValueExpr.withValue(
-                      PrimitiveValue.builder().setValue("0").setType(TypeNode.INT).build()))
-              .build());
-      // TODO(summerji): Remove the following two lines.
-      // This temporary workaround will be removed soon, so it doesn't need a test.
-      setLocalVariableExpr(varExpr.toBuilder().setIsDecl(false).build());
+    GeneralForStatement build() {
+      GeneralForStatement generalForStatement = autoBuild();
+      VariableExpr localVarExpr = generalForStatement.initializationExpr().variableExpr();
+      // Declare a variable inside for-loop initialization expression.
+      if (localVarExpr.isDecl()) {
+        Preconditions.checkState(
+            localVarExpr.scope().equals(ScopeNode.LOCAL),
+            String.format(
+                "Variable %s declare in a general for-loop cannot have a non-local scope",
+                localVarExpr.variable().identifier().name()));
+        Preconditions.checkState(!localVarExpr.isStatic(), "Modifier 'static' not allow here.");
+      }
+      // TODO (unsupport): Uncomment the following code if public setter for the initialization,
```

Comment:
In general, checking in commented-out unused code is discouraged because it may suffer "code rot" as the codebase evolves. Instead, could you please convert these into a brief summary on the conditions to check instead?

Reply:
Make sense. Updated the comments.